src/test/java/org/mule/extension/db/lifecycle/OracleArtifactLifecycleListenerTestCase.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2023 Salesforce, Inc. All rights reserved.
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.extension.db.lifecycle;
+
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.hasItem;
+import static org.hamcrest.Matchers.hasProperty;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.isIn;
+import static org.hamcrest.core.IsNot.not;
+
+import org.mule.extension.db.internal.lifecycle.OracleArtifactLifecycleListener;
+import org.mule.sdk.api.artifact.lifecycle.ArtifactLifecycleListener;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+import org.hamcrest.Matcher;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+@RunWith(Parameterized.class)
+public class OracleArtifactLifecycleListenerTestCase extends AbstractArtifactLifecycleListenerTestCase {
+
+  public static final String DRIVER_PACKAGE = "oracle.jdbc";
+  public static final String DRIVER_TIMER_THREAD = "oracle.jdbc.diagnostics.Diagnostic.CLOCK";
+
+  public OracleArtifactLifecycleListenerTestCase(String groupId, String artifactId, String version) {
+    super(groupId, artifactId, version);
+  }
+
+  @Parameterized.Parameters
+  public static Collection<Object[]> parameters() {
+    return Arrays.asList(new Object[][] {
+        {"com.oracle.database.jdbc", "ojdbc8", "23.2.0.0"}});
```

Comment:
The idea is just to test with one version, do you think could be worth it to test with a least version 21 and 22 ?

Reply:
Good point. I’ve concentrated our efforts primarily on the latest versions and those with known issues, as these are most critical for our current objectives. In doing so, I’ve structured the code to be easily extendable for additional versions when needed.
However, it's important to note that expanding to other versions isn't straightforward due to the variances between each version. Specifically, we would need to adjust our assertions to accommodate the differences in each version or range of versions. This adds a layer of complexity, as we must tailor our testing approach to align with the unique characteristics of each version.
Moving forward, while we have the framework in place to incorporate more versions, we should be mindful of the additional considerations and resources this expansion may require.